### PR TITLE
feat(bootstrap): allow prepending managed dotfile lines

### PR DESCRIPTION
**Pull request by jdx**
## Summary

- Add `position = "prepend"` for exact-line dotfile edits while keeping `append` as the default.
- Leave an existing exact match in place instead of moving or duplicating it.
- Preserve all unrelated target bytes, including CRLF line endings and a UTF-8 BOM.
- Document the `/etc/zshrc` use case and validate the new field in the schema.

## Example

```toml
[dotfiles]
"/etc/zshrc/zdotdir" = { line = 'ZDOTDIR=$HOME/.config/zsh/', position = "prepend" }
```

## Verification

- `MBX_DISABLE=1 cargo test --locked --bin mise system::edits::tests`
- `MBX_DISABLE=1 mise run test:e2e e2e/cli/test_dotfiles_edits e2e/config/test_schema_tombi`
- `MBX_DISABLE=1 mise run lint`

The Cargo wrapper emitted action-prediction warnings and stalled around the final test-binary handoff during the initial run. The exact checks passed with the repository-documented `MBX_DISABLE=1` bypass.

*AI-assisted — Tool: Codex; model: unavailable; version: unavailable.*

## Summary by CodeRabbit

* **New Features**
  * Added `position = "prepend"` support for dotfile line edits.
  * Missing lines can be inserted at the beginning or end of a file; appending remains the default.
  * Insertion preserves newline styles and leading file markers.

* **Bug Fixes**
  * Improved handling of BOM-prefixed and BOM-only files.
  * Line insertion occurs only when exact content is absent.
  * Added validation for unsupported placement options and edit types.

* **Documentation**
  * Updated dotfile editing documentation and configuration schema to describe line placement behavior.

---

> [!NOTE]
> **Medium Risk**
> Changes byte-level file editing for dotfile line inserts (BOM/CRLF), which can affect bootstrap-managed config files if edge cases are wrong, but scope is limited to the edit engine with strong e2e tests.
> 
> **Overview**
> Dotfile **line** edits gain optional **`position = "prepend"`** (default **`append`**) so a missing exact line can be inserted at the start of the file—documented for cases like setting `ZDOTDIR` in `/etc/zshrc/zdotdir`.
> 
> Line apply/unapply in `edits.rs` was reworked so insertion only happens when the line is absent, existing matches are not moved or duplicated, and unrelated bytes stay intact (**CRLF** endings, leading **UTF-8 BOM**). Unapply picks the first or last matching line depending on prepend vs append. Config parsing rejects **`position`** on blocks and unknown placement values; the JSON schema and tombi tests enforce **`position`** only on line entries.
> 
> E2E coverage adds prepend/BOM scenarios and updates dotfiles docs for the new semantics.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit ff37ecffba8b5d8f817fa47ef6292e933bd2ec15. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: src/system/edits.rs [hunk 1]
```diff
@@ -1123,19 +1158,34 @@ fn apply_to_string(req: &EditRequest, desired: Option<&str>, text: &str) -> Resu
...
             // an earlier entry in the same batch may have just written an
             // identical line (two ids, same text) — stay idempotent against
             // the file's current content, not the state at plan time
-            if !lines.iter().any(|l| l == line) {
-                lines.push(line.clone());
+            let (bom, body) = text
+                .strip_prefix('\u{feff}')
+                .map_or(("", text), |body| ("\u{feff}", body));
+            if body.lines().any(|candidate| candidate == line) {
+                return Ok(text.to_string());
```
[the remark this answers is not in the document]
Comment (jdx) [reply src/system/edits.rs:1189]: Fixed in 401a43c39. Precheck now strips a leading UTF-8 BOM before exact-line comparison, with regression coverage for an already-present first line.

*AI-assisted — Tool: Codex; model: unavailable; version: unavailable.*

File: src/system/edits.rs [hunk 2]
```diff
@@ -1123,19 +1158,34 @@ fn apply_to_string(req: &EditRequest, desired: Option<&str>, text: &str) -> Resu
...
+            let newline = body
+                .find('\n')
+                .filter(|&i| i.checked_sub(1).and_then(|i| body.as_bytes().get(i)) == Some(&b'\r'))
+                .map_or("\n", |_| "\r\n");
+            let out = match position {
+                LinePosition::Prepend if body.is_empty() => format!("{bom}{line}{newline}"),
+                LinePosition::Prepend => format!("{bom}{line}{newline}{body}"),
+                LinePosition::Append if text.is_empty() => format!("{line}{newline}"),
+                LinePosition::Append if text.ends_with('\n') => format!("{text}{line}{newline}"),
+                LinePosition::Append => format!("{text}{newline}{line}{newline}"),
```
[the remark this answers is not in the document]
Comment (jdx) [reply src/system/edits.rs]: Fixed in 401a43c39. BOM-only files now use the logical body emptiness check and retain the BOM at byte zero without inserting a blank line; the e2e test compares exact bytes.

*AI-assisted — Tool: Codex; model: unavailable; version: unavailable.*

File: schema/mise.json [hunk 3]
```diff
@@ -4173,7 +4173,20 @@
...
+                "description": "where to insert a missing line",
+                "anyOf": [
+                  {
+                    "enum": ["append", "prepend"]
+                  },
+                  {
+                    "type": "string"
+                  }
+                ]
               },
```
[the remark this answers is not in the document]
Comment (jdx) [reply schema/mise.json:4190]: Fixed in 401a43c39. The schema now requires line for the branch that permits position, and a Tombi regression rejects block plus position.

*AI-assisted — Tool: Codex; model: unavailable; version: unavailable.*

File: src/system/edits.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (jdx) [reply src/system/edits.rs:1189]: Fixed across 401a43c39 and 118f73ca6. Precheck and unapply planning compare the BOM-stripped logical first line, while removal excludes the BOM byte span. The regression verifies forced unapply restores BOM plus CRLF content byte-for-byte.

*AI-assisted — Tool: Codex; model: unavailable; version: unavailable.*

[the remark this answers is not in the document]
Comment (jdx) [reply src/system/edits.rs:1189]: Fixed across 401a43c39 and 118f73ca6. BOM handling is now consistent in precheck, apply, unapply planning, and byte-span removal; BOM-only append also treats the logical body as empty. Regression coverage checks both application and unapplication byte-for-byte.

*AI-assisted — Tool: Codex; model: unavailable; version: unavailable.*

File: schema/mise.json [hunk 4]
```diff
@@ -4190,14 +4203,23 @@
...
+                  "position": false,
                   "template": false,
                   "comment": false
                 }
               },
               {
+                "required": ["line"],
                 "properties": {
                   "content": false
                 }
```
[the remark this answers is not in the document]
Comment (jdx) [reply schema/mise.json:4221]: Fixed in 3a19f3df3. The line-operation schema branch now rejects both source and block, and the Tombi regression covers each mixed form plus position without line.

*AI-assisted — Tool: Codex; model: unavailable; version: unavailable.*

File: schema/mise.json [hunk 5]
```diff
@@ -4190,13 +4203,24 @@
...
                   "comment": false
                 }
               },
               {
+                "required": ["line"],
                 "properties": {
-                  "content": false
+                  "content": false,
+                  "source": false,
+                  "block": false
```
[the remark this answers is not in the document]
Comment (jdx) [reply schema/mise.json]: Fixed in ff37ecffb. The line-operation schema now rejects mode, exclude, template, and comment in addition to the existing whole-file/block fields. The Tombi regression loop covers every rejected combination.

*AI-assisted — Tool: Codex; model: unavailable; version: unavailable.*
